pkg/kds/global/components_test.go

```diff
@@ -70,6 +71,38 @@ var _ = Describe("Global Sync", func() {
 		closeFunc()
 	}
 
+	VerifyResourcesWereNotSynchronizedToGlobal := func() {
+		dp := &mesh_proto.Dataplane{
+			Networking: &mesh_proto.Dataplane_Networking{
+				Address: "192.168.0.1",
+				Inbound: []*mesh_proto.Dataplane_Networking_Inbound{{
+					Port: 1212,
+					Tags: map[string]string{
+						mesh_proto.ZoneTag:    "kuma-cluster-1",
+						mesh_proto.ServiceTag: "service-1",
+					},
+				}},
+				Outbound: []*mesh_proto.Dataplane_Networking_Outbound{
+					{
+						Tags: map[string]string{
+							mesh_proto.ServiceTag:  "web",
+							mesh_proto.ProtocolTag: "http",
+						},
+					},
+				},
+			},
+		}
+		err := zoneStores[0].Create(context.Background(), &mesh.DataplaneResource{Spec: dp}, store.CreateByKey("dp-1", "mesh-1"))
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() int {
+			actual := mesh.DataplaneResourceList{}
+			err := globalStore.List(context.Background(), &actual)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
nit: I'd inject Gomega to eventually if you use Expect. Just like you did for components_test